## Final Project Submission

Please fill out:
* Student name: Lucas Wilkerson
* Student pace: Flex/part time
* Scheduled project review date/time: 
* Instructor name: Morgan Jones
* Blog post URL:


# Home Price Analysis 

In [ ]:
# Add Graphic 

## Project Overview 

For this project, I sought to explore how certain housing characteristics influence the price of a home in King county. Specifically, I used linear regression to show how specific housing characteristics impact the price of the home.

## Business Problem and Stakeholder

A real estate company in King County wants to increase customer acquisition and retention by providing transparent and useful information regarding the sales prices of homes. Utilizing this data, they can assist customers/clients that are both looking to buy or sell a home, understand what to budget for with a new home or what to expect to sell their current home for. For the project specifically we will be exploring which housing characteristics are the most important and have the biggest impact on sales price.

## Data Understanding 

In [2]:
# Importing standard Packages 
import pandas as pd
import numpy as np
import math 

# Importing packages for visualization 
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

# Importing packages for statistics
import statsmodels.api as sm
from scipy import stats
from sklearn.metrics import mean_absolute_error, mean_squared_error
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import OneHotEncoder
from sklearn.model_selection import train_test_split

## Data Preparation

In [3]:
#Loading dataset
kc_house_df = pd.read_csv("data/kc_house_data.csv")
kc_house_df.head(10)

id        date      price  bedrooms  bathrooms  sqft_living  \
0  7399300360   5/24/2022   675000.0         4        1.0         1180   
1  8910500230  12/13/2021   920000.0         5        2.5         2770   
2  1180000275   9/29/2021   311000.0         6        2.0         2880   
3  1604601802  12/14/2021   775000.0         3        3.0         2160   
4  8562780790   8/24/2021   592500.0         2        2.0         1120   
5  2807100156   7/20/2021   625000.0         2        1.0         1190   
6  5122400111  11/17/2021  1317227.0         3        3.0         2080   
7  7137850210   4/28/2022   820000.0         3        2.5         2214   
8  2944500680   3/17/2022   780000.0         4        2.5         2340   
9  2619950340   6/21/2021   975000.0         4        2.5         2980   

   sqft_lot  floors waterfront greenbelt  ... sewer_system sqft_above  \
0      7140     1.0         NO        NO  ...       PUBLIC       1180   
1      6703     1.0         NO        NO  ...       PUBLIC       1570   
2      6156     1.0         NO        NO  ...       PUBLIC       1580   
3      1400     2.0         NO        NO  ...       PUBLIC       1090   
4       758     2.0         NO        NO  ...       PUBLIC       1120   
5      5688     1.0         NO        NO  ...       PUBLIC       1190   
6     27574     1.0         NO        NO  ...      PRIVATE       2080   
7      3506     2.0         NO        NO  ...       PUBLIC       2214   
8      8125     2.0         NO        NO  ...       PUBLIC       2340   
9      5859     2.0         NO        NO  ...       PUBLIC       2980   

  sqft_basement sqft_garage sqft_patio yr_built  yr_renovated  \
0             0           0         40     1969             0   
1          1570           0        240     1950             0   
2          1580           0          0     1956             0   
3          1070         200        270     2010             0   
4           550         550         30     2012             0   
5             0         300          0     1948             0   
6             0           0        150     1951             0   
7             0         440        206     2019             0   
8             0         440         70     1989             0   
9             0         540        170     2011             0   

                                             address        lat        long  
0  2102 Southeast 21st Court, Renton, Washington ...  47.461975 -122.190520  
1  11231 Greenwood Avenue North, Seattle, Washing...  47.711525 -122.355910  
2  8504 South 113th Street, Seattle, Washington 9...  47.502045 -122.225200  
3  4079 Letitia Avenue South, Seattle, Washington...  47.566110 -122.290200  
4  2193 Northwest Talus Drive, Issaquah, Washingt...  47.532470 -122.071880  
5  1602 North 185th Street, Shoreline, Washington...  47.763470 -122.340155  
6  2633 Southwest 164th Place, Burien, Washington...  47.455470 -122.367220  
7  24913 122nd Place Southeast, Kent, Washington ...  47.378355 -122.178625  
8  2721 Southwest 343rd Place, Federal Way, Washi...  47.293770 -122.369320  
9  27950 Northeast 147th Circle, Duvall, Washingt...  47.733170 -121.965305  

[10 rows x 25 columns]

In [4]:
kc_house_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30155 entries, 0 to 30154
Data columns (total 25 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             30155 non-null  int64  
 1   date           30155 non-null  object 
 2   price          30155 non-null  float64
 3   bedrooms       30155 non-null  int64  
 4   bathrooms      30155 non-null  float64
 5   sqft_living    30155 non-null  int64  
 6   sqft_lot       30155 non-null  int64  
 7   floors         30155 non-null  float64
 8   waterfront     30155 non-null  object 
 9   greenbelt      30155 non-null  object 
 10  nuisance       30155 non-null  object 
 11  view           30155 non-null  object 
 12  condition      30155 non-null  object 
 13  grade          30155 non-null  object 
 14  heat_source    30123 non-null  object 
 15  sewer_system   30141 non-null  object 
 16  sqft_above     30155 non-null  int64  
 17  sqft_basement  30155 non-null  int64  
 18  sqft_g

Above I have read in the file and previewed the head of the dataset along with utilizing .info() to get a general idea of the dataset. We have a total of over 30,000 data points with very low amounts of missing values. With low amounts of missing values, I plan to drop all rows with missing values. It looks like we have 25 columns that have types that include integers, floats, and objects. Looking at the columns and data types it seems we have several numerical values and catergorical values. I may need to transform some variables but that will be determined as I explore more into the data.

In [5]:
#Dropping all rows with null values and dropping duplicate values/rows
kc_house_df.dropna(inplace= True)
kc_house_df = kc_house_df.drop_duplicates()
kc_house_df.drop(["id"], axis = 1, inplace = True)

kc_house_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 30110 entries, 0 to 30154
Data columns (total 24 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   date           30110 non-null  object 
 1   price          30110 non-null  float64
 2   bedrooms       30110 non-null  int64  
 3   bathrooms      30110 non-null  float64
 4   sqft_living    30110 non-null  int64  
 5   sqft_lot       30110 non-null  int64  
 6   floors         30110 non-null  float64
 7   waterfront     30110 non-null  object 
 8   greenbelt      30110 non-null  object 
 9   nuisance       30110 non-null  object 
 10  view           30110 non-null  object 
 11  condition      30110 non-null  object 
 12  grade          30110 non-null  object 
 13  heat_source    30110 non-null  object 
 14  sewer_system   30110 non-null  object 
 15  sqft_above     30110 non-null  int64  
 16  sqft_basement  30110 non-null  int64  
 17  sqft_garage    30110 non-null  int64  
 18  sqft_p

Duplicates and null values were dropped along with the id column as this column is not needed. Called .info() to confirm changes. We now have a total of 30,110 entries regarding home sales with 24 columns. 

## Data Analysis

## Modeling 

## Regression Results 

## Conclusion/ Recommendations

## Limitations